optax/_src/wrappers_test.py

```diff
@@ -394,6 +394,32 @@ def test_mask_fn(self):
     np.testing.assert_allclose(updates['b'][1],
                                grads['b'][1] + 0.1*params['b'][1])
 
+  @chex.all_variants
+  @parameterized.named_parameters(
+      ('scale', lambda: transform.scale(-1.)),  # stateless test
+      ('sgd', _build_sgd),  # stateful test
```

Comment:
Totally agree! I forgot that SGD no longer has state by default after #54. 

I think `_build_simple_adam` is meant to be used as a [stateful version of `_build_sgd`](https://github.com/deepmind/optax/blob/1cb88b56c87319b4d342dfeaceef29f8217cf3f6/optax/_src/wrappers_test.py#L34-L40), however, that is actually incorrect: `_build_simple_adam` is signed gradient descent, not sgd.

It happens to work properly in the [one test it is used in](https://github.com/deepmind/optax/blob/1cb88b56c87319b4d342dfeaceef29f8217cf3f6/optax/_src/wrappers_test.py#L70-L120) because the gradient values are +/- 1.

Example illustrating that it is indeed not equivalent to sgd in general:

```py
import jax
import jax.numpy as jnp
import optax

params = jnp.full(1, 2.0)
grads = jnp.full_like(params, 3.0)

sgd = optax.sgd(1.0)
sgd_state = sgd.init(params)
updates, _ = sgd.update(grads, sgd_state)
print('SGD Update:', updates)

adam = optax.adam(1.0, b1=0., b2=0.)
adam_state = adam.init(params)
updates, _ = adam.update(grads, adam_state)
print('Adam Update: ', updates)
```
Output:
```
>> SGD Update: [-3.]
>> Adam Update:  [-1.]
```

I think we should replace `_build_simple_adam` with `_build_stateful_sgd` where the difference is `momentum=0.0`.  Then, I can fix tests that use `_build_sgd` and `_build_simple_adam` as well as add the test you suggested.

How does that sound?

Reply:
Hi! Yes, that sounds good to me! Let's do it in a separate PR so that we can roll back the changes separately if necessary. Would you like to do the PR for that or shall I? (I'm happy either way, just want to make sure we don't work on it at the same time)

Reply:
I'll open a PR!

Reply:
Done.